ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -485,6 +485,104 @@ abstract class EventStorageBackendTemplate(
+               event_offset > p.participant_all_divulged_contracts_pruned_up_to_inclusive -- TODO pruning check
+               OR p.participant_all_divulged_contracts_pruned_up_to_inclusive IS NULL
+             )
+             AND event_offset <= $pruneUpToInclusive
+             -- Only prune create events which did not have a locally hosted party before their creation offset
+             AND NOT EXISTS (
+              SELECT 1
+                FROM party_entries p
+               WHERE p.typ = 'accept'
+                 AND p.ledger_offset <= c.event_offset
+                 AND #${queryStrategy.isTrue("p.is_local")}
+                 AND #${queryStrategy.arrayContains("c.flat_event_witnesses", "p.party_id")}
+             )
+             $pruneAfterClause
+         """
+      } else cSQL""""""
+
+    pruneWithLogging("prune all")(SQL"""
+       WITH archived_events AS (
+         SELECT contract_id, event_sequential_id
+           FROM participant_events_consuming_exercise
+          WHERE event_offset <= $pruneUpToInclusive
+       ),
+       created_events_to_be_deleted AS (
+          SELECT contract_id, event_sequential_id, transaction_id FROM participant_events_create creates_to_be_deleted
+                WHERE creates_to_be_deleted.contract_id IN (SELECT contract_id FROM archived_events)
+          ${unionWithSelectImmediateDivulgenceCreatedEvents()} -- TODO pruning: Merge in where clause
+       ),
+       creates_distinct_transaction_ids_to_be_pruned AS (
+         -- TODO pruning: Might not be needed
+         SELECT DISTINCT(transaction_id) AS transaction_id FROM created_events_to_be_deleted
+       ),
+       nonconsuming_events_to_be_deleted AS (
+          SELECT contract_id, event_sequential_id FROM participant_events_non_consuming_exercise nonconsuming_to_be_deleted
+            WHERE nonconsuming_to_be_deleted.contract_id IN (SELECT contract_id FROM archived_events)
+       ),
+       deleted_retroactive_divulgence AS(
+         -- TODO redundant if re-applied for all divulgence events pruning
+         -- Retroactive divulgence events (only for contracts archived before the specified offset)
+         DELETE FROM participant_events_divulgence delete_events
+               WHERE delete_events.contract_id IN (SELECT contract_id FROM archived_events)
+                 AND delete_events.event_offset <= $pruneUpToInclusive -- TODO pruning: do we even need this guard? Probably not
+       ),
+       deleted_create_events AS (
+         -- Create events (only for contracts archived before the specified offset)
+         DELETE FROM participant_events_create delete_events
+               WHERE delete_events.contract_id IN (SELECT contract_id FROM created_events_to_be_deleted)
+       ),
+       deleted_nonconsuming AS (
+         -- Non-consuming exercises
+         DELETE FROM participant_events_non_consuming_exercise delete_events
+            WHERE delete_events.contract_id IN (SELECT contract_id FROM archived_events)
```

Comment:
Shouldn't all non-consuming before the pruning offset get deleted?

Reply:
Yep, answered above